plugins/routed-eni/cni.go

```diff
@@ -182,6 +180,15 @@ func add(args *skel.CmdArgs, cniTypes typeswrapper.CNITYPES, grpcClient grpcwrap
 	return cniTypes.PrintResult(result, cniVersion)
 }
 
+// generateHostVethName returns a name to be used on the host-side veth device.
+// The veth name is generated such that it aligns with the value expected
+// by Calico for NetworkPolicy enforcement.
+func generateHostVethName(namespace, podname string) string {
+	h := sha1.New()
+	h.Write([]byte(fmt.Sprintf("%s.%s", namespace, podname)))
```

Comment:
It looks like you're embedding metadata about the pod into the interface name here.  If this is purely to assure name uniqueness, this is fine, but if Calico relies on this data for mapping Network Policy, we should have a stronger contract.  At minimum, documentation about the data embedding requirement and constraints here; but preferably an API to map interface to k8s metadata.

Reply:
Are we going to force Calico with this AWS plugin? I hope we are on the direction that different network policy implementation can work with this plugin.

Reply:
@lilida We definitely want to support multiple deployment scenarios.